galaxy_importer/collection.py

```diff
@@ -77,6 +84,24 @@ def _import_collection(file, filename, logger, cfg):
     return attr.asdict(data)
 
 
+def _extract_tar_shell(filepath, extract_dir):
+    cwd = os.path.dirname(os.path.abspath(filepath))
+    file_name = os.path.basename(filepath)
+    args = [
+        'tar',
+        f'--directory={extract_dir}',
+        '-xf',
+        file_name,
+    ]
+    try:
+        proc = run(args, cwd=cwd, stderr=PIPE)
+    except (FileNotFoundError, SubprocessError) as e:
+        raise exc.ImporterError(f'Error in tar extract subprocess: {str(e)}')
+    if proc.returncode != 0:
```

Comment:
Minor: You could use `run` with `check=True` argument. In this case `run` raises an exception in case of non zero exit code.

Reply:
updated